frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,238 @@ pub struct PoolRoles<AccountId> {
 	pub state_toggler: Option<AccountId>,
 }
 
+/// Pool commission.
+///
+/// The pool `root` can set commission configuration after pool creation. By default, all commission
+/// values are `None`. Pool `root`s can set `max` and `throttle` configurations before setting an
+/// initial commission `current` value - the commission percentage and payee of commission.
+///
+/// The `max` commission value can only be decreased after the initial value is set, to prevent
+/// commission from repeatedly increasing.
+///
+/// An optional commission throttle allows the pool to set strict limits to how much commission can
+/// change in each update, and how often updates can take place. If a `throttle` is set before a
+/// commission `current` is set, the initial commission `current` value will not be subject to
+/// throttling. Subsequent commission updates will be.
+#[derive(
+	Encode, Decode, DefaultNoBound, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value can
+	/// only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configuration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Commission<T> {
+	/// Get the current commission percentage of this pool.
+	///
+	/// Returns zero if commission has not yet been configured.
+	fn commission_or_zero(&self) -> Perbill {
+		self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero())
```

Comment:
```suggestion
		self.current
			.as_ref()
			.map(|(x, _)| *x.min(&GlobalMaxCommission::<T>::get().unwrap_or(Bounded::max_value())))
			.unwrap_or(Perbill::zero())
```

then we won't need any permissionless operation to enforce the global max either.

Reply:
Pretty cool, we might be able to get away with only doing this is on `do_reward_payout`. This function is currently only being used in `throttling()`.

Reply:
Applied here: https://github.com/paritytech/substrate/pull/12511#discussion_r1057434892